internal/provider/webhook_recipient_resource.go

```diff
@@ -420,3 +500,32 @@ func validateAttributesWhenTemplatesIncluded(ctx context.Context, data models.We
 
 	}
 }
+
+func validateAttributesWhenVariablesIncluded(ctx context.Context, data models.WebhookRecipientModel, resp *resource.ValidateConfigResponse) {
+	var templates []models.WebhookTemplateModel
```

Comment:
nit: given that we're now decoding templates in this function now as well it might be worth just moving the logic from both of these validate helpers right into `ValidateConfig`